cfpb/cfgov-refresh
# Re-work video module to use composition instead of inheritance

This PR re-works the video player module to remove the inheritance relationship between `VideoPlayer` and `YoutubePlayer` and instead creates a utility module `youtube-api` that's composed into the `VideoPlayer` in an attempt to simplify the video player codebase and solve https://[GHE]/CFGOV/platform/issues/3456 

## Changes

- Remove inheritance of YoutubePlayer and VideoPlayer and use VideoPlayer directly.
- Include video `<iframe …>` directly in the template instead of creating it dynamically via JS.
- Set default export on js-loader utility.
- Separate out FeaturedContentModule script from VideoPlayer.

## Testing

1. Pull branch
1. `./frontend.sh`
1. `gulp test:unit` should pass.
1. Visit and test play a bunch of different video/FCM pages:
   a. Standard static image FCM: http://localhost:8000/practitioner-resources/your-money-your-goals/
   b. Video FCM: http://localhost:8000/prepaid-rule/
   c. Standalone video: http://localhost:8000/data-research/prepaid-accounts/issuer-instructions/
   d. Event video: http://localhost:8000/about-us/events/archive-past-events/fall-2018-bcfp-advisory-committee-meetings-washington-dc/

## Notes

- FCM supersedes https://github.com/cfpb/cfgov-refresh/pull/5161
